hedera-node/hedera-app/src/main/java/com/hedera/node/app/signatures/crypto/SignatureMapAccessor.java

```diff
@@ -0,0 +1,100 @@
+ * {@code 0x01ff}. Then the minimum information Alice needs to send is just which unique prefix
+ * corresponds to which signature.
+ *
+ * <p>This class handles the problem of finding the signature bytes for a given public key prefix
+ * inside a {@code SignatureMap}.
+ *
+ * <p>There is an important exception in which public key prefixes are not sufficient, and the
+ * network <i>does</i> need the full public key. This is when the public key will be needed to give
+ * permissions to an EVM system contract. In this case, the network cannot possibly predict which
+ * public key will be needed, so the user must provide it explicitly.
+ */
+@SuppressWarnings({"java:S1068", "java:S1192"})
+public class SignatureMapAccessor {
+    private static final byte[] MISSING_SIG = new byte[0];
+
+    private final SignatureMap signatureMap;
+
+    public SignatureMapAccessor(final SignatureMap signatureMap) {
+        this.signatureMap = signatureMap;
+    }
+
+    /**
+     * Returns the signature bytes for the given public key prefix, <b>if</b> the {@code
+     * SignatureMap} contains a unique {@code SignaturePair} with that prefix.
+     *
+     * <p>If there is no such {@code SignaturePair}, this returns an empty byte array.
+     *
+     * <p>If there are multiple {@code SignaturePair} whose prefixes match the given key, throws a
+     * {@link NonUniquePrefixException}.
+     *
+     * @param publicKey the public key that should match exactly one prefix in the map
+     * @return the signature bytes for the given public key, or an empty byte array if no prefixes
+     *     match
+     * @throws NonUniquePrefixException if more than one prefix matches
+     */
+    public byte[] getSignatureByPublicKey(final byte[] publicKey) throws NonUniquePrefixException {
+        throw new AssertionError("Not implemented");
+    }
+
+    /**
+     * Returns the signature bytes from the first {@code SignaturePair} whose full public key is set
+     * in the prefix; and would have the given address on the Ethereum network.
+     *
+     * <p>(Recall that on Ethereum, the address of an account is the last 20 bytes of the Keccak-256
+     * hash of the public key.)
+     *
+     * @param evmAddress an EVM address
+     * @return the signature bytes for the full public key that implies the given address
+     */
+    public byte[] getSignatureByEvmAddress(final byte[] evmAddress) {
+        throw new AssertionError("Not implemented");
+    }
+
+    /**
+     * Following zero or more calls to {@link #getSignatureByPublicKey(byte[])} and {@link
+     * #getSignatureByEvmAddress(byte[])}, this method returns a list of all the {@code
+     * SignaturePair}'s <b>with full-key prefixes</b> that were not used in those calls.
+     *
+     * @return the remaining {@code SignaturePair}'s with full-key prefixes
+     */
+    public List<SignaturePair> getRemainingExplicitSignatures() {
```

Comment:
This method came somewhat surprising to me. Nothing previously indicated, that `SignatureMapAccessor` tracks which signatures have been used already. I think this functionality should be made visible much more prominently, maybe even in the class name.